launcher.py

```diff
@@ -407,107 +422,122 @@ def ensure_MediaProcessor_build(system, re_build=False):
+            ),
+            DependencyHandler(
+                "Python Dependencies",
+                check_cmd={"all": [check_python_dependecies_installed]},
+                install_cmd={
+                    "Windows": [f"{str(Utils.get_virtualenv_folder()/ 'pip.exe')} install -r requirements.txt"],
+                    "all": [f"{str(Utils.get_virtualenv_folder()/ 'pip')} install -r requirements.txt"],
+                },
+            ),
+        ]
+        self.system = system
+        self.URL = "http://127.0.0.1:8080"
+        self.tries = 5
+        self.timeout = 0.5
+        self.setup(log_level, log_file)
+
+    def setup(self, log_level: str, log_file: bool):
+        """
+        Installs the required dependencies and Setup Configuration for the web application.
+        """
+        self.config = Config(self.system, log_level, log_file)
+        for dependency in self.required_dependencies:
+            dependency.ensure_installed(self.system)
 
+    def run(self):
+        try:
+            python_path = Utils.get_virtualenv_folder() / ("python.exe" if self.system == "Windows" else "python")
 
-def log_stream(stream, log_function):
-    """Logs output from a stream."""
-    for line in iter(stream.readline, ""):
-        log_function(line.strip())
-    stream.close()
+            # Start the backend
+            app_process = subprocess.Popen(
+                [python_path, "app.py"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True,
+            )
+            atexit.register(app_process.terminate)
 
+            # Threads to handle stdout and stderr asynchronously
+            threading.Thread(target=log_stream, args=(app_process.stdout, logging.debug), daemon=True).start()
+            threading.Thread(target=log_stream, args=(app_process.stderr, logging.debug), daemon=True).start()
 
-def launch_web_application(system):
-    try:
-        python_path = get_virutalenv_folder() / ("python.exe" if system == "Windows" else "python")
+            # Give the process some time to initialize
+            time.sleep(0.5)
 
-        # Start the backend
-        app_process = subprocess.Popen(
-            [python_path, "app.py"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            text=True,
-        )
-        atexit.register(app_process.terminate)
+            # Check if the process is still running
+            if app_process.poll() is not None:
+                error_output = app_process.stderr.read()
+                logging.error(f"Error starting the backend: {error_output}")
+                sys.exit(1)
 
-        # Threads to handle stdout and stderr asynchronously
-        threading.Thread(target=log_stream, args=(app_process.stdout, logging.debug), daemon=True).start()
-        threading.Thread(target=log_stream, args=(app_process.stderr, logging.debug), daemon=True).start()
+            webbrowser.open("http://127.0.0.1:8080")
```

Comment:
port should be parametric as we may provide it as a config field for the user in the future.